src/main/java/uk/gov/companieshouse/accounts/association/controller/AssociationsListForCompanyController.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.companieshouse.accounts.association.controller;
+
+import java.util.Objects;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.accounts.association.AccountsAssociationServiceApplication;
+import uk.gov.companieshouse.accounts.association.exceptions.BadRequestRuntimeException;
+import uk.gov.companieshouse.accounts.association.service.AssociationsService;
+import uk.gov.companieshouse.api.accounts.associations.api.AssociationsListForCompanyInterface;
+import uk.gov.companieshouse.api.accounts.associations.model.AssociationsList;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@RestController
+public class AssociationsListForCompanyController implements AssociationsListForCompanyInterface {
+
+    private final AssociationsService associationsService;
+
+    private static final Logger LOG = LoggerFactory.getLogger( AccountsAssociationServiceApplication.applicationNameSpace );
+
+    public AssociationsListForCompanyController(AssociationsService associationsService) {
+        this.associationsService = associationsService;
+    }
+
+    @Override
+    public ResponseEntity<AssociationsList> getAssociationsForCompany( final String companyNumber, final String xRequestId, final Boolean includeRemoved, final Integer pageIndex, final Integer itemsPerPage ) {
+
+        LOG.debug( String.format( "%s: Attempting to fetch users that are associated with company %s. includeRemoved=%b, itemsPerPage=%d, and pageIndex=%d.", xRequestId, companyNumber, includeRemoved, itemsPerPage, pageIndex ) );
+
+        if ( Objects.isNull( xRequestId ) ){
+            LOG.error( "xRequestId was not provided" );
+            throw new BadRequestRuntimeException( "Please check the request and try again" );
+        }
+
+        if ( pageIndex < 0 ){
+            LOG.error( "pageIndex was less then 0" );
+            throw new BadRequestRuntimeException( "Please check the request and try again" );
+        }
+
+        if ( itemsPerPage <= 0 ){
+            LOG.error( "itemsPerPage was less then 0" );
+            throw new BadRequestRuntimeException( "Please check the request and try again" );
+        }
```

Comment:
also this? where did this requirement come from

Reply:
If you pass an itemsPerPage that is <= 0, then PageRequest.of() will crash.

Reply:
again dud to test it manually, please showcase this to me by running locally.

Reply:
Same comment as the last one